Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -543,6 +552,69 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable {
             }
         }
 
+        public object this[[NotNull]PythonTuple index] {
+            get {
+                CheckBuffer();
+
+                int[] tupleValues = FixTuple(index);
+
+                int actualIndex = 0;
+                for (int i = 0; i < index.Count; i++) {
+                    actualIndex *= (int)shape[i];
```

Comment:
Tuple access has some overhead to it, maybe cache the value `var item = (int)shape[i];` instead of calling it twice? If you wanted to go further (probably not worth it), since I think we fully control shape we could directly access the underlying array.